docs/book/src/topics/clusterclass.md

```diff
@@ -0,0 +1,88 @@
+2. AzureManagedClusterTemplate
+3. AzureManagedControlPlaneTemplate
+
+Each resource is a template for the corresponding CAPZ resource. For example, the AzureClusterTemplate is a template for the CAPZ AzureCluster resource. The template contains a set of parameters that are able to be shared across multiple clusters.
+
+## Deploying a Self-Managed Cluster with ClusterClass
+
+To deploy a self-managed cluster with ClusterClass, you must first create a ClusterClass resource. The ClusterClass resource defines the cluster topology, including the control plane and machine deployment templates. The ClusterClass resource also defines the parameters that can be used to customize the cluster topology. 
+
+Please refer to the Cluster API book for more information on how to write a ClusterClass topology: https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/write-clusterclass.html
+
+For a self-managed cluster, the AzureClusterTemplate defines the Azure infrastructure for the cluster. The following example shows a basic AzureClusterTemplate resource:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterTemplate
+metadata:
+  name: capz-clusterclass-cluster
+  namespace: default
+spec:
+  template:
+    spec:
+      location: westus2
+      networkSpec:
+        subnets:
+        - name: control-plane-subnet
+          role: control-plane
+        - name: node-subnet
+          natGateway:
+            name: node-natgateway
+          role: node
+      subscriptionID: 00000000-0000-0000-0000-000000000000
+```
+
+## Deploying a Managed Cluster (AKS) with ClusterClass
+
+Deploying an AKS cluster with ClusterClass is similar to deploying a self-managed cluster. Instead of using the AzureClusterTemplate, you must use both an AzureManagedClusterTemplate and AzureManagedControlPlaneTemplate. Due to the nature of managed Kubernetes and the control plane implementation, the infrastructure provider (and therefore the AzureManagedClusterTemplate) for AKS cluster is basically a no-op. The AzureManagedControlPlaneTemplate is used to define the AKS cluster configuration, such as the Kubernetes version and the number of nodes. 
+
+The following example shows a basic AzureManagedClusterTemplate and AzureManagedControlPlaneTemplate resource:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureManagedClusterTemplate
+metadata:
+  name: capz-clusterclass-cluster
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureManagedControlPlaneTemplate
+metadata:
+  name: capz-clusterclass-control-plane
+spec:
+  location: westus2
+  subscriptionID: 00000000-0000-0000-0000-000000000000
+  version: 1.25.2
+```
+
+## Excluded Fields
+
+Since a ClusterClass is a template for a Cluster, there are some fields that are not allowed to be shared across multiple clusters. For each of the ClusterClass resources, the following fields are excluded:
+
+### AzureClusterTemplate
+- `spec.resourceGroup`
```

Comment:
Is there a better place we can refer to these fields? Or do we need to document them explicitly at all? Listing them in the docs like this is sure to go stale, especially has we continue to add features.

Reply:
I think we can just leave it out, since users can refer to the CRDs for available fields